### ISSUE TITLE
I use optuna to select model params and "num_features_to_select" of select_features function. The result is bad.

### ISSUE THREAD

**Issue by fengshansi** [catboost/catboost#2674]
```
def objective(trial: optuna.trial.Trial):

    model_params = {
        "learning_rate": trial.suggest_float("learning_rate", 0.001, 0.1, log=True),
        "depth": trial.suggest_int("depth", 1, 10),
        "subsample": trial.suggest_float("subsample", 0.05, 1.0),
        "colsample_bylevel": trial.suggest_float("colsample_bylevel", 0.05, 1.0),
        "min_data_in_leaf": trial.suggest_int("min_data_in_leaf", 1, 100),
    }

    features_select_params = {
        "num_features_to_select": trial.suggest_int("num_features_to_select")
    }
```
For 1 trial. I first select the features, then train a model. However, compared to only optuna model params, I cannot improve my metric. Should I only optuna model params, and then start a new trial optuna program to optuna  num_features_to_select?
